Add support for listing invites (aka pending members)

abl opened the pull request:
### Checklist
- [X] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [X] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [X] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] I've updated the documentation if necessary.

### Motivation and Context
It's currently possible to send invites using Portal but there's no way to view the status of invites that are in flight. Combined with a group of administrators onboarding new users this can lead to a certain amount of duplication.

These changes were tested against a live account; basic rspec coverage was added as well.

### Description
This PR, building on #8047, wraps the `/getInvites` endpoint and emits `Invite` objects.

I purposely kept invited users separate from actual users; an argument could be made that invited users should be shown in `Spaceship::Portal::Persons.all` but the available metadata for an `Invite` differs from the metadata of a `Person`. It might make sense to move this out to `Spaceship::Portal::Invites` (along with `Spaceship::Portal::Persons.invite`) but I wanted to get feedback first.

dateJoined in Person is a formatted string; dateCreated/dateExpires comes back as a timestamp. For consistency, this PR alters Person to parse a Time object as well and updates the relevant specs.

abl committed:
Parse Person join date for consistency.

abl committed:
Add the ability to list ADC invites (pending members).

abl committed:
Update invite and person specs.

abl committed:
Consistent capitalization.

KrauseFx commented on spaceship/lib/spaceship/portal/invite.rb:
```diff
@@ -0,0 +1,43 @@
+
+      class << self
+        def factory(attrs)
+          # rubocop:disable Style/RescueModifier
```
It would be great if we can avoid using inline `rescue` without an error type specification. Is `attrs['dateCreated']` `nil` sometimes? If so, can we check of `nil` values instead, or catch only the exception type we expect here?

KrauseFx commented on spaceship/lib/spaceship/portal/person.rb:
```diff
@@ -42,6 +42,15 @@ def remove!
+
+      class << self
+        def factory(attrs)
+          # rubocop:disable Style/RescueModifier
```
Same here, ideally we don't want to disable that `rubocop` rule

KrauseFx commented on spaceship/spec/portal/invite_spec.rb:
```diff
@@ -0,0 +1,21 @@
+    }
+    invited = Spaceship::Portal::Invite.factory(attrs)
+    # Roles are downcased to match Person
+    expect(invited.type).to eq("admin")
```
Great tests ❤️

KrauseFx reviewed (changes requested):
Hey @abl, thanks for the pull request, this looks great 👍 The only thing that we'd love to have changed before merging the change is to remove all inline `rescue` blocks. Let me know if my comments make sense.

taquitos commented on spaceship/spec/portal/fixtures/inviteList.json:
```diff
@@ -0,0 +1,32 @@
+    {
+      "inviteId": "A2B4C6D8EF",
+      "recipientRole": "MEMBER",
+      "inviterName": "Joshua Liebowitz",
```
best inviter 😄

taquitos commented:
Woooo this is awesome!

abl committed:
Remove inline rescue; update specs to cover error cases.

abl commented:
Thanks :)

I copied the rescue pattern from [certificate.rb:231](https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/portal/certificate.rb#L231-L235) - in this case, the intent is that if Apple gives us back a new date format we can still return the raw API value.

I've rewritten the inline rescues as standard rescues and added specs to cover - it feels like I'm missing a better way to write this kind of capture, though?

KrauseFx committed:
Fix spelling

abl commented:
Meta - I'm looking at the raw patches and I'm not seeing any issues on authorship. @KrauseFx, you were already in the Google organization when you committed, right? If so, we should point the Googlebot team at this PR.

abl commented:
(also, thanks for the typo fix!)

abl commented:
Sorry, had to rebase and pull this branch forward (we're using it internally) and it looks like that broke the review.

KrauseFx reviewed (approved):
This is great, thanks 👍
